### PR TITLE
Add grenade cooking

### PR DESCRIPTION
**Pull request by deadlap**

### CODE REVIEW

**Comment by mtib** [comment]
That's called cooking.

File: src/game/states/play.rs [hunk 1]
```diff
@@ -531,7 +536,7 @@ impl GameState for Play {
                 }
             }
             Mouse(MouseButton::Right) => {
-                if let Some(gm) = self.world.player.wep.utilities.throw_grenade(ctx, &mut s.mplayer).unwrap() {
+                if let Some(gm) = self.world.player.wep.utilities.cock_grenade(ctx, &mut s.mplayer).unwrap() {
```
Comment (mtib) [inline src/game/states/play.rs]: cook, not cock.

Comment (mtib) [reply src/game/states/play.rs]: At least it's consequently called `cock` everywhere..

Comment (deadlap) [reply src/game/states/play.rs]: Falch said that was what it was called...

File: src/obj/grenade.rs [hunk 2]
```diff
@@ -101,6 +109,29 @@ impl Grenade {
         let expl_img = (a.get_img(ctx, "weapons/explosion1")).clone();
         Mesh::from_raw(ctx, &vertices, &indices, Some(expl_img))
     }
+    pub fn explode(obj: &mut Object, palette: &Palette, grid: &Grid, player: &mut Player, enemies: &mut [Enemy]) -> GrenadeUpdate {
```
Comment (LFalch) [inline src/obj/grenade.rs]: obj doesn't really have to be mutable.

File: src/obj/grenade.rs [hunk 3]
```diff
@@ -162,7 +173,19 @@ impl Grenade {
         }
         GrenadeUpdate::None
     }
-
+    pub fn update_cooked(ctx: &mut Context, obj: &mut Object, fuse: &mut f32, palette: &Palette, grid: &Grid, player: &mut Player, enemies: &mut [Enemy]) -> GrenadeUpdate {
+        obj.pos = player.obj.pos + 20. * angle_to_vec(player.obj.rot);
+        if *fuse > DELTA {
+            *fuse -= DELTA;
+            if (*fuse) > DELTA && !mouse::button_pressed(ctx, MouseButton::Right) {
```
Comment (LFalch) [inline src/obj/grenade.rs]: You don't need to check whether fuse is above DELTA again here.

File: src/game/states/play.rs [hunk 4]
```diff
@@ -456,7 +461,7 @@ impl GameState for Play {
         let img = s.assets.get_img(ctx, "common/crosshair");
         graphics::draw(ctx, &*img, drawparams)
     }
-    fn event_up(&mut self, s: &mut State, ctx: &mut Context, event: Event) {
+    fn event_down(&mut self, s: &mut State, ctx: &mut Context, event: Event) {
```
Comment (LFalch) [inline src/game/states/play.rs]: This changes the behaviour the of everything else as well. Is this really what we want?

**Review by LFalch — CHANGES_REQUESTED** [review]
Much of the fuse logic is the same for Fused and Thrown, maybe that can be put together in a function?

**Comment by LFalch** [comment]
This should close #35

**Pull request closed without merging**